cmd/monaco/download/downloaders_test.go

```diff
@@ -77,3 +77,94 @@ func TestGetDownloaderPanic(t *testing.T) {
 		getDownloader[v2.ClassicApiType](downloaders)
 	})
 }
+
+func Test_prepareAPIs(t *testing.T) {
+	t.Run("no endpoint marked as 'skip' is present", func(t *testing.T) {
+		tests := []struct {
+			name  string
+			given downloadConfigsOptions
+		}{
+			{
+				name:  "onlyAutomation",
+				given: downloadConfigsOptions{onlyAutomation: true},
+			},
+			{
+				name:  "onlySettings",
+				given: downloadConfigsOptions{onlySettings: true},
+			},
+			{
+				name:  "onlyAPIs",
+				given: downloadConfigsOptions{onlyAPIs: true},
+			},
+			{
+				name:  "specificAPIs is marked as 'skip'",
+				given: downloadConfigsOptions{specificAPIs: []string{"extension"}},
+			},
+			{
+				name: "without special cases",
+			},
+		}
+
+		for _, tc := range tests {
+			t.Run(tc.name, func(t *testing.T) {
+				apis := prepareAPIs(tc.given)
+				for _, e := range apis {
+					assert.False(t, e.SkipDownload)
+				}
+			})
+		}
+	})
+
+	t.Run("filtration of deprecated", func(t *testing.T) {
+		tests := []struct {
+			name       string
+			given      downloadConfigsOptions
+			deprecated bool
+		}{
+			{
+				name:       "onlyAutomation",
+				given:      downloadConfigsOptions{onlyAutomation: true},
+				deprecated: false,
+			},
+			{
+				name:       "onlySettings",
+				given:      downloadConfigsOptions{onlySettings: true},
+				deprecated: false,
+			},
+			{
+				name:       "onlyAPIs",
+				given:      downloadConfigsOptions{onlyAPIs: true},
+				deprecated: false,
+			},
+			{
+				name:       "specificAPIs is marked with 'deprecatedBy'",
```

Comment:
nit
```suggestion
				name:       "specificAPI marked with 'deprecatedBy' is not filtered out",
```

I think the name could be improved to make the meaning of the test case clearer. 
Compare the test method names in downloader_test that you've removed. I find those make it clearer what assumption is actually covered by the testcase